src/rules/camelcase.rs

```diff
@@ -36,6 +36,36 @@ impl LintRule for Camelcase {
     visitor.visit_program(program, program);
     visitor.report_errors();
   }
+
+  fn docs(&self) -> &'static str {
+    r#"Enforces the use of camelCase in variable names
+
+Consistency in a code base is key for readability and maintainability.  This rule
+enforces variable definitions and assignments to be in camelCase.  Of note:
+* `_` is allowed at the start or end of a variable
+* All uppercase variable names (e.g. constants) may have `_` in their name
+* Function calls are exempt
+* This rule also applies to variables imported via ES6 module imports
+    
+### Invalid:
+```typescript
+let first_name = "Ichigo";
+function do_something(){}
+import { not_camelCased } from "external-module";
```

Comment:
```suggestion
let first_name = "Ichigo";
const obj = { last_name: "Hoshimiya" };
const { last_name } = obj;

function do_something() {}
function foo({ snake_case = "default value" }) {}

class snake_case_class {}

import { not_camelCased } from "external-module.js";
export * as not_camelCased from 'mod.ts';
```

This rule is quite complicated, so it would be nice to provide plenty of invalid examples.

Reply:
Happy to add more examples.  Would your destructuring example be valid if `obj` came from an external library outside your control?

Reply:
> No, because you would need to alias the destructured variable.

exactly 👍